### PR TITLE
Add fee middleware test suite functions (E2E #5)

### PR DESCRIPTION
**Pull request by chatton**
## Description

This PR adds some functionality that will be used in the coming fee middleware E2E tests.

* The ability to broadcast arbitrary messages for our test suites.
  * RegisterCounterPartyPayee
  * PayPacketFeeAsync

* Initialize GRPC clients for test chains.
* Query GRPC endpoints
  * QueryCounterPartyPayee  
  * QueryIncentivizedPacketsForChannel

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/fee_middleware_test.go [hunk 1]
```diff
@@ -18,13 +24,51 @@ type FeeMiddlewareTestSuite struct {
 	testsuite.E2ETestSuite
 }
 
+// RegisterCounterPartyPayee broadcasts a MsgRegisterCounterpartyPayee message.
+func (s *FeeMiddlewareTestSuite) RegisterCounterPartyPayee(ctx context.Context, chain *cosmos.CosmosChain, user broadcast.User, portID, channelID, relayerAddr, counterpartyPayeeAddr string) (sdk.TxResponse, error) {
```
Comment (crodriguezvega) [inline e2e/fee_middleware_test.go:28]: With so many arguments the line becomes quite long, should we split it in multiple lines for better readability?

**Review by crodriguezvega — APPROVED** [review]

File: e2e/fee_middleware_test.go [hunk 2]
```diff
@@ -18,13 +24,51 @@ type FeeMiddlewareTestSuite struct {
...
+	})
+
+	if err != nil {
+		return "", err
+	}
+	return res.CounterpartyPayee, nil
+}
+
+// PayPacketFeeAsync broadcasts a MsgPayPacketFeeAsync message.
+func (s *FeeMiddlewareTestSuite) PayPacketFeeAsync(ctx context.Context, chain *cosmos.CosmosChain, user broadcast.User, packetID channeltypes.PacketId, packetFee feetypes.PacketFee) (sdk.TxResponse, error) {
```
Comment (damiannolan) [inline e2e/fee_middleware_test.go:48]: ditto on @crodriguezvega's point

File: e2e/fee_middleware_test.go [hunk 3]
```diff
@@ -18,13 +24,51 @@ type FeeMiddlewareTestSuite struct {
...
+	msg := feetypes.NewMsgPayPacketFeeAsync(packetID, packetFee)
+	return s.BroadcastMessages(ctx, chain, user, msg)
+}
+
+// QueryIncentivizedPacketsForChannel queries the incentivized packets on the specified channel.
+func (s *FeeMiddlewareTestSuite) QueryIncentivizedPacketsForChannel(ctx context.Context, chain *cosmos.CosmosChain, portId, channelId string) ([]*feetypes.IdentifiedPacketFees, error) {
```
Comment (damiannolan) [inline e2e/fee_middleware_test.go:54]: ditto on @crodriguezvega's point

File: e2e/testsuite/testsuite.go [hunk 4]
```diff
@@ -29,13 +36,18 @@ const (
 // E2ETestSuite has methods and functionality which can be shared among all test suites.
 type E2ETestSuite struct {
 	suite.Suite
+	grpcClientSets map[string]GRPCClientSet
```
Comment (damiannolan) [inline e2e/testsuite/testsuite.go:39]: mega nit but should we add a line break after `suite.Suite`...?

File: e2e/testsuite/testsuite.go [hunk 5]
```diff
@@ -29,13 +36,18 @@ const (
...
 	paths          map[string]path
 	logger         *zap.Logger
 	DockerClient   *dockerclient.Client
 	network        string
 	startRelayerFn func(relayer ibc.Relayer)
 }
 
+type GRPCClientSet struct {
```
Comment (damiannolan) [inline e2e/testsuite/testsuite.go:47]: nit: I think we should add godocs to all exported types

File: e2e/testsuite/testsuite.go [hunk 6]
```diff
@@ -196,6 +225,37 @@ func (s *E2ETestSuite) GetChainBNativeBalance(ctx context.Context, user *ibctest
...
+	)
+	s.Require().NoError(err)
+	s.T().Cleanup(func() {
+		if err := grpcConn.Close(); err != nil {
+			s.T().Logf("failed closing GRPC connection to chain %s: %s", chain.Config().ChainID, err)
+		}
+	})
+
+	if s.grpcClientSets == nil {
+		s.grpcClientSets = map[string]GRPCClientSet{}
```
Comment (damiannolan) [inline e2e/testsuite/testsuite.go:251]: Any reason that we should not use `make(map[string]GRPCClientSet)` ?

File: e2e/testsuite/testsuite.go (hunk 5 above)
Comment (damiannolan) [reply e2e/testsuite/testsuite.go:47]: What's the reasoning behind the naming of having the `Set` suffix? Initially when I read this I thought it was supposed to be a set data structure but I'm slightly confused

Comment (damiannolan) [reply e2e/testsuite/testsuite.go:47]: Yep sounds good to me! :) Thanks 🤝

**Review by damiannolan — APPROVED** [review]

**Pull request merged**